### ISSUE TITLE
expfmt should parse HELP text

### ISSUE THREAD

**Issue by gouthamve** [prometheus/common#143]
See: https://github.com/prometheus/prometheus/issues/4602

promtool is using expfmt https://github.com/prometheus/prometheus/blob/master/util/promlint/promlint.go#L68 which doesn't check the HELP text.

**Comment by brian-brazil** [comment]
No, the issue here is the ToUpper on https://github.com/prometheus/common/blob/master/expfmt/text_parse.go#L507
